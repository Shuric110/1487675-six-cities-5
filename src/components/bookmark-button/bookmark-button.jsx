import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";

import {authorizationStatusPropType, offerPropType} from "../../props";
import {AsyncActionCreator} from "../../store/async-action";


const BookmarkButton = (props) => {
  const {children, baseClassName, offer: {id, isFavorite}, setIsFavorite} = props;
  return (
    <button className={`${baseClassName} button ${isFavorite ? `${baseClassName}--active` : ``}`} type="button"
      onClick={() => setIsFavorite(id, !isFavorite)}
    >
      {children}
    </button>
  );
};

BookmarkButton.propTypes = {
  children: PropTypes.node.isRequired,
  baseClassName: PropTypes.string.isRequired,
  offer: offerPropType.isRequired,
  authorizationStatus: authorizationStatusPropType.isRequired,
  setIsFavorite: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  authorizationStatus: state.USER.authorizationStatus
});

const mapDispatchToProps = (dispatch) => ({
  setIsFavorite(offerId, isFavorite) {
    dispatch(AsyncActionCreator.setIsFavorite(offerId, isFavorite));
  },
});

export {BookmarkButton};
export default connect(mapStateToProps, mapDispatchToProps)(BookmarkButton);
