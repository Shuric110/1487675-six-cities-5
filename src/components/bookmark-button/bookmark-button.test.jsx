import React from "react";
import renderer from "react-test-renderer";
import {BookmarkButton} from "./bookmark-button";

const noop = () => {};

const offerWithFavoriteFlag =
{
  id: 1,
  city: {
    id: 1,
    name: `Paris`,
    coordinates: {
      latitude: 48.8606146,
      longitude: 2.3354553
    },
    zoom: 12,
  },
  pictures: [
    `/img/room.jpg`,
  ],
  placePicture: `/img/room.jpg`,
  coordinates: {
    latitude: 52.369553943508,
    longitude: 4.85309666406198
  },
  mapZoom: 12,
  isPremium: false,
  isFavorite: true,
  nightlyCost: 80,
  title: `Wood and stone place`,
  type: `room`,
  rating: 4.2,
  description: `Excellent choice for travelers, offering many helpful amenities designed to enhance your stay.`,
  bedrooms: 1,
  maxAdults: 2,
  features: [
    `Wi-Fi`,
  ],
  host: {
    name: `Angelina`,
    avatar: `/img/avatar-angelina.jpg`,
    isSuper: true
  },
};

const offerWithoutFavoriteFlag = Object.assign({}, offerWithFavoriteFlag, {isFavorite: false});

describe(`Render BookmarkButton`, () => {
  it(`BookmarkButton for favorite order renders correctly`, () => {
    const tree = renderer
      .create(
          <BookmarkButton
            baseClassName="class1"
            offer={offerWithFavoriteFlag}
            setIsFavorite={noop}
          >
            <div className="test1">test1</div>
          </BookmarkButton>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`BookmarkButton for not favorite order renders correctly`, () => {
    const tree = renderer
      .create(
          <BookmarkButton
            baseClassName="class2"
            offer={offerWithoutFavoriteFlag}
            setIsFavorite={noop}
          >
            <div className="test2">test2</div>
            <div className="test3">test3</div>
          </BookmarkButton>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });


});
