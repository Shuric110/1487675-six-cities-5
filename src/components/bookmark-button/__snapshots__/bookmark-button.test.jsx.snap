// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render BookmarkButton BookmarkButton for favorite order renders correctly 1`] = `
<button
  className="class1 button class1--active"
  onClick={[Function]}
  type="button"
>
  <div
    className="test1"
  >
    test1
  </div>
</button>
`;

exports[`Render BookmarkButton BookmarkButton for not favorite order renders correctly 1`] = `
<button
  className="class2 button "
  onClick={[Function]}
  type="button"
>
  <div
    className="test2"
  >
    test2
  </div>
  <div
    className="test3"
  >
    test3
  </div>
</button>
`;
